### PR TITLE
1.0.6 readme update

### PR DESCRIPTION
**Pull request by jondade**
A reworking of the documentation to break them up into smaller chunks. Hopefully easier to digest that way.

### CODE REVIEW

File: Documentation/CONFIGURATION.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+* [Prerequisites](#prerequisites)
+* [Configure the Fastly Service](#configure-the-fastly-service)
+* [Basic Configuration](#configure-the-module)
+* [Further Configuration Options](#further-configuration-options)
+
+## Prerequisites
+
+To proceed with configuration you will need to have
+[signed up for Fastly's service](https://www.fastly.com/signup). You will need
+the API key (or a user key with purge access). Details of how to find these
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: Recommend API Tokens?

File: Documentation/CONFIGURATION.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+have [signed up](https://www.fastly.com/signup) for the service. Once you have
+confirmed the account and logged in the Fastly application will take you to a
+wizard to configure your first service. This is further documented on [Fastly's documentation](https://docs.fastly.com/guides/basic-setup/sign-up-and-create-your-first-service).
+
+You will need to email Fastly's support team to
+[request access](https://docs.fastly.com/guides/vcl/uploading-custom-vcl) to
+upload VCL to your account.
+Once this is enabled refresh the Fastly application and navigate to the VCL
+page in the latest unlocked version of the service. Once there upload the
+[Fastly_Cdn VCL](../etc/fastl.vcl) as the main VCL. Review the generated VCL
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: That should be `fastly.vcl`

File: Documentation/CONFIGURATION.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+to ensure that it is correctly implemented. Then activate the new version.
+
+## Configure the Module
+
+This section will describe how to input the API key and service ID into the
+Fastly_Cdn module. It assumes that the module has already been installed
+without error. If not please see [installing the module](INSTALLATION.md) for
+details of how to do this.
+
+To enter the credentials for the Fastly_Cdn module to use open a browser and
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: Turn the "to use" into a comma?

File: Documentation/CONFIGURATION.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+Expand the 'Full Page Cache' section then choose 'Fastly CDN' in the 'Caching
+Application' dropdown.
+
+Adjust the TTL for public content to suit the needs of the server. (This value
+is in seconds.)
+
+Expand the 'Fastly Configuration' section to enter the remaining necessary
+details.
+
+Leave the 'Fastly API Endpoint' set as the default value.
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: Perhaps remove this configuration option?

File: Documentation/CONFIGURATION.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+serve stale content after the "normal" TTL has expired.
+
+### Stale content delivery time in case of backend error
+
+This setting defines the time in seconds that stale content can be delivered in
+case the backend is down or cannot respond properly.
+
+### Purge category
+
+This option binds automatic purge of category (Fastly) cache with is update
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: I'm not sure what this means

File: Documentation/CONFIGURATION.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+This option binds automatic purge of CMS page (Fastly) cache with its update
+event. If set to "Yes" CMS page cache is invalidated each time CMS page update
+event occurs (i.e. CMS page content update via Magento admin).
+
+### Use Soft Purge
+
+Using soft purge will not remove the content immediately from Fastly's cache
+but mark it as stale. In combination with the stale timings your customers will
+be serverd stale content very fast while Fastly is updating the content in the
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: `serverd` typo

File: Documentation/CONFIGURATION.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+background.
+
+### GeoIP handling
+
+GeoIP handling will get the country of a web client
+based on it's IP address. This feature can be used to either automatically
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: `it's` should be `its`

File: Documentation/CONFIGURATION.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+In your Magento2 backend go to Stores -> Configuration -> System in the
+"Advanced" section and open "Full Page Cache" tab and choose "Yes" with "Enable
+GeoIP". Make sure to be on store configuration scope.
+
+#### General behavior
+
+The Fastly module supports two options to serve the store view based on GeoIP
+country code: It can show the visitor a modal dialog to give him the option to
+switch to a frontend that fits to his country or he can be redirect
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: `redirect` should be `redirected`

File: Documentation/CONFIGURATION.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+automatically. Both actions are performed using JavaScript and can be adjusted
+if necessary.
+
+In the VCL the cookie headers are examined. If either the X-Magento-Vary or
```
Comment (acme) [inline Documentation/CONFIGURATION.md]: Use `` to show that these headers are code?

File: Documentation/INSTALLATION.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+- [Prerequisites](#prerequisites)
+- [Installation](#installation)
+- [Troubleshooting](#troubleshooting)
+
+## Prerequisites
+
+Before installing the Fastly_Cdn module you should setup a
+test environment as you will need to put Fastly in front which will certainly
+take a while for configuring and testing. If you directly rollout this solution
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `rollout` → `roll out`

File: Documentation/INSTALLATION.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+to your production server you might experience issues that could affect your
+normal eBusiness.
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `eBusiness` → `business`
(every business is an eBusiness)

File: Documentation/INSTALLATION.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+problems in your environment as debugging Magento2 issues with Fastly in front
+might be difficult.
+
+Fastly_Cdn supports Magento2 Community and Enterprise Edition from version 2.0
+onwards.
+
+You need an account with [fastly.com](https://www.fastly.com/signup) which allows
+[uploading of custom VCL](https://docs.fastly.com/guides/vcl/uploading-custom-vcl).
+If you need professional services for assistance with setting up your
+environment please contact fastly.com.
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `fastly.com` → `support@fastly.com`

File: Documentation/INSTALLATION.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+The installation of the Magento module is pretty easy and can be performed in
+a few ways.
+
+- [Install from Magento Marketplace](#installing-from-the-magento-marketplace-using-web-setup-wizard)
+- [Install using Composer](#installing-using-composer)
+- [Install from Zip file](#installing-from-zip-file)
+
+#### Installing from the Magento Marketplace using Web Setup Wizard
+
+This will require an account with Magento Commerce and the api keys will be
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `api` → `API`

Comment (acme) [inline Documentation/INSTALLATION.md]: What API keys?

File: Documentation/INSTALLATION.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+1. Open a browser to the [Magento Marketplace](https://marketplace.magento.com/fastly-magento2.html)
+    and add the module to the cart. Check out and ensure that this is added to
+    your account.
+1. Log into the admin section of the Magento system in which to install the
+    module as an administrator.
+1. Start the Web Setup Wizard by navigating to 'System > Web Setup Wizard'.
+1. Click 'Component Manager' to synchronise with Magento Marketplace.
+1. Click to 'Enable' the Fastly_Cdn module. This will start the wizard.
+1. Follow the on screen instructions ensuring to create backups.
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `ensuring` → `, being sure`

File: Documentation/INSTALLATION.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+    ```
+    bin/magento cache:clean
+    ```
+
+1. Once this has completed log in to the Magento Admin panel and proceed to
+    [Configuring the Module](CONFIGURATION.md).
+
+#### Installing from zip file
+
+1. Open a browser to [Github](https://github.com/fastly/fastly-magento2/releases)
```
Comment (acme) [inline Documentation/INSTALLATION.md]: `Github` → `GitHub`

File: Documentation/OPENING-ISSUES.md [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+Issues are for tracking broken functionality and are intended for managing
+development work to address these.
+
+If you have an issue with usage or need general assistance using this module,
+or Fastly in general please use the [support infomation](../README.md#support).
+
+This describes what content should be added to a issue to help with an
+expedient resolution.
+
+What to post...
```
Comment (acme) [inline Documentation/OPENING-ISSUES.md]: This should be a header?

File: Documentation/OTHER-FUNCTIONS.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+## Contents
+
+* [Purging](#purging)
+* [VCL Design Exceptions](#vcl-design-exceptions)
+
+## Purging
+
+Fastly caches objects for a certain period of time according to their TTL.
+After that the object will not be requested from the web server or Magento
```
Comment (acme) [inline Documentation/OTHER-FUNCTIONS.md]: Well, as long as it is hot

File: Documentation/OTHER-FUNCTIONS.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+again. Until the TTL expires Fastly will deliver the cached object no matter
+what will change within Magento or the webserver's file system. To force Fastly
+to cleanup its' cache and to retrieve the information again from the backend
```
Comment (acme) [inline Documentation/OTHER-FUNCTIONS.md]: `its'` → its

File: Documentation/OTHER-FUNCTIONS.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+modified without the need to invalidate any other object which will save a lot
+of resources on high frequented stores.
+
+### Purge by store
+
+You can purge objects in Fastly's Cache based on the store by just clicking
+"Clean Fastly CDN Cache by store". This will only remove content that is
+generated from Magento. Images, CSS files or JavaScripts will not be purged.
+
+### Purge a URI
```
Comment (acme) [inline Documentation/OTHER-FUNCTIONS.md]: The header says `URI`, but the text says `URL`. Seeing as the application uses `URL`, change the heading to `URL`?

File: Documentation/OTHER-FUNCTIONS.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+It is also possible to purge a single url (e.g. page) using "Quick Purge".
```
Comment (acme) [inline Documentation/OTHER-FUNCTIONS.md]: `url` → `URL`

File: Documentation/README.md [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+
+## Contents
+
+* [Installation instructions](INSTALLATION.md)
+* [Configuration instructions](CONFIGURATION.md)
+* [Purging and Other Functions](OTHER-FUNCTIONS.md)
+* [Troubleshooting instructions](TROUBLESHOOTING.md)
+* [How to report bugs](OPENING-ISSUES.md)
+
+Please report any errors or problems with these documents as github issues.
```
Comment (acme) [inline Documentation/README.md]: `github` → `GitHub`

File: README.md [hunk 22]
```diff
@@ -1,247 +1,81 @@
-**FASTLY CDN FOR MAGENTO2 DOCUMENTATION**
+# FASTLY CDN FOR MAGENTO2 DOCUMENTATION
 
-Thank you for using the "Fastly CDN module for Magento2" (FastlyCDN).
+Thank you for using the "Fastly CDN module for Magento2" (Fastly_Cdn).
 
 This package contains everything you need to connect fastly.com (Fastly) with
 your Magento commerce shop and to get the most out of Fastly's powerful caching
-capabilities for a blazing fast eCommerce site. The FastlyCDN module consists of two main components:
```
Comment (acme) [inline README.md]: `FastlyCDN` → `Fastly_Cdn`

File: README.md [hunk 23]
```diff
@@ -1,247 +1,81 @@
...
-
-Ensure that your Magento2 store is running without any
-problems in your environment as debugging Magento2 issues with Fastly in front
-might be difficult.
-
-FastlyCDN supports Magento2 Community and Enterprise Edition from version 2.0 onwards.
-
-You need an account with [fastly.com](https://www.fastly.com/signup) which allows
-[uploading of custom VCL](https://docs.fastly.com/guides/vcl/uploading-custom-vcl). If you need
-professional services for setup your environment please contact fastly.com.
```
Comment (acme) [inline README.md]: `fastly.com` → `support@fastly.com`

File: README.md [hunk 24]
```diff
@@ -1,247 +1,81 @@
...
-to "Yes" and category cache will be invalidated each time a category update occurs.
+For Fastly customers with a [Support Package](https://www.fastly.com/support)
+please reach out via the normal channels.
 
-### Purge product
-This option binds purge of product (Fastly) cache with product and  product's stock
-update. If set to "Yes" product pages cache is
-invalidated each time product update or product's stock update occurs.
+If there are issues/errors with integrating the module, please post
+[details](Documentation/OPENING-ISSUES.md) in the github repository issues.
```
Comment (acme) [inline README.md]: `github` → `GitHub`

File: README.md [hunk 25]
```diff
@@ -1,247 +1,81 @@
...
-# 5. VCL Design Exceptions
-
-By default Varnish does not take into account
-User-Agent string of a request when building its cache object. Magento Design
-Exceptions use regular expressions to match different design configurations to
-User-Agent strings. In order to make Design Exceptions work with Varnish you
-will have to renew Varnish VCL each time Design Exceptions are updated.
-To do this use the button "Export VCL for Varnish fastly" button in the configuration section and upload the new VCL
-to the Fastly service.
+- Fork the repository from the master branch.
```
Comment (acme) [inline README.md]: Move these out to a CONTRIBUTING.md?

File: Documentation/CONFIGURATION.md (hunk 1 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: There's no UI for tokens yet, so I'd like to keep this simple. That can be changed and recommended when there's a UI to help.

File: Documentation/CONFIGURATION.md (hunk 2 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Yes. Yes it should. :)

File: Documentation/CONFIGURATION.md (hunk 4 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Required to pass Magento module certification, so need to document not to change it.

File: Documentation/CONFIGURATION.md (hunk 6 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Agreed.

File: Documentation/CONFIGURATION.md (hunk 5 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Modified.

File: Documentation/CONFIGURATION.md (hunk 7 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Yes it should ;)

File: Documentation/CONFIGURATION.md (hunk 8 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: yup

File: Documentation/CONFIGURATION.md (hunk 9 above)
Comment (jondade) [inline Documentation/CONFIGURATION.md]: Agreed.

File: Documentation/INSTALLATION.md (hunk 10 above)
Comment (jondade) [inline Documentation/INSTALLATION.md]: rolled out

File: Documentation/INSTALLATION.md (hunk 11 above)
Comment (jondade) [inline Documentation/INSTALLATION.md]: S'pose

File: Documentation/INSTALLATION.md (hunk 13 above)
Comment (jondade) [inline Documentation/INSTALLATION.md]: Added clarification

File: Documentation/INSTALLATION.md (hunk 15 above)
Comment (jondade) [inline Documentation/INSTALLATION.md]: Yup.

File: Documentation/OPENING-ISSUES.md (hunk 16 above)
Comment (jondade) [inline Documentation/OPENING-ISSUES.md]: Makes sense.

File: Documentation/OTHER-FUNCTIONS.md (hunk 19 above)
Comment (jondade) [inline Documentation/OTHER-FUNCTIONS.md]: Yup and updated for consistent URLs

File: Documentation/README.md (hunk 21 above)
Comment (jondade) [inline Documentation/README.md]: yup

File: README.md (hunk 22 above)
Comment (jondade) [inline README.md]: Line was removed.

File: README.md (hunk 24 above)
Comment (jondade) [inline README.md]: yup.

File: README.md (hunk 25 above)
Comment (jondade) [inline README.md]: Makes sense. Done.

**Comment by acme** [comment]
Reviewed and approved.

**Pull request merged**